miden-tx/src/compiler/mod.rs

```diff
@@ -51,6 +51,12 @@ impl TransactionCompiler {
         }
     }
 
+    pub fn with_debug_mode(mut self, in_debug_mode: bool) -> Self
+    {
```

Comment:
nit: the formatting seems a bit off here (I'm surprised the CI didn't catch it).

Also, since this is a part of the public interface, let's add a doc comment to this function.